Use fasttransform

RensDimmendaal opened the pull request.

Commit:
use fasttransform

Commit:
make show_batch fasttransform compatible

Commit:
fastcore.typedispatch -> plum.dispatch

Commit:
add setups to docs, not fasttransform related but broke running of nbdev_test

Commit:
more moves from fastcore.typedispatch -> plum.dispatch

Commit:
updated prints w new Transform repr

RensDimmendaal commented on fastai/data/core.py:
```diff
@@ -31,7 +31,12 @@ def show_batch(
 
 # %% ../../nbs/03_data.core.ipynb 10
-@typedispatch
+def get_show_batch_func(x_typ=Any, y_typ=Any, samples_typ=Any):
```
@jph00 is this what you had in mind when you said 'make a show_batch_basic' func?
I couln't do exactly that because sometimes the x_typ!=object.

jph00 replied:
How about then adding `show_batch_basic(...)` which calls `get_show_batch_func(object)(...)`? And maybe rename `get_show_batch_func` to `get_show_batch`?

jph00 reviewed (commented):
This looks great! You'll need to add `fasttransforms` to settings.ini too.

jeraymond committed:
Fix #4072 untar_data() ignores provided base

Updated untar_data() to pass a None config to FastDownload when an overridden
base is provided. FastDownload _always_ uses a provided config object and
ignores base if both are given.

Commit:
update fastcore requirement to >=1.8.0,<1.9

Commit:
add fasttransform as dependency

jph00 committed:
Merge branch 'main' into use-fasttransform
